### ISSUE TITLE
Add $resolved property to $firebaseObject and $firebaseArray

### ISSUE THREAD

**Issue by soumak77** [firebase/angularfire#836]
I would like to be able to have a simple boolean value tell me whether the data has been loaded for a `$firebaseObject` and `$firebaseArray`.  It would be similar to how [ng-resource](https://docs.angularjs.org/api/ngResource/service/$resource) uses the `$resolved` property to provide this information.

I know I can use the `$loaded()` function to do something once the data is loaded, but having a `$resolved` property would make it a lot easier for data-binding once the data is loaded.

**Comment by soumak77** [comment]
I was able to get the solution working with `$resolved` as a function, but not as a property.

**Comment by jwngr** [comment]
Thanks for the issue and PR. Can you explain the use case here? Using a code example would be very helpful.

**Comment by katowulf** [comment]
This can be accomplished using $loaded. There's no need to store a promise as this already returns one.

```
var list = $angularFire( ... );
var $resolved = list.$loaded();
```

**Comment by soumak77** [comment]
The `$resolved` method returns a boolean indicating whether the promise has completed.  The use case is identical to the use case for why the `$resolved` property exists for `$resource`.  `$resource` provides a `$promise` and `$resolved` properties which mirror the firebase `$loaded()` and `$resolved()` functions.  It avoids having to write the following boiler plate code over and over again throughout an application:

```javascript
$scope.resolved = false;
$scope.data = $firebaseObject(...);
$scope.data.$loaded().finally(function() {
  $scope.resolved = true;
});
```
```html
<div ng-if="resolved">
  ...
</div>
```

Using the new method:
```javascript
$scope.data = $firebaseObject(...);
```
```html
<div ng-if="data.$resolved()">
  ...
</div>
```

**Comment by katowulf** [comment]
This should work for you.

```
app.factor('ArrayWithResolved', function($firebaseArray) {
   function ArrayWithResolved() {
       var self = this;
       this.$resolved = false;
       $firebaseArray.apply(this, arguments);
       this.$loaded().then(function() {
            self.$resolved = true;
       });
       return this.$list;
   });
   return $firebaseArray.$extend(ArrayWithResolved);
});
```

**Comment by soumak77** [comment]
@katowulf I was hoping to not have to do this for every firebase object/array I create.  The `$resolved` property has proven to be extremely useful when loading data through ng-resource so I was hoping to have similar functionality from firebase.  I understand it's duplicate functionality as the developer can get the same results, but the it avoids so much redundant code.

Just my two cents.  I'll add it manually as you suggested if it doesn't make it's way into the core library.

**Comment by katowulf** [comment]
I'm not very familiar with ng-resource. Can you provide some examples?

I'll leave this open for discussion and reopen the PR if others feel this is useful.

It's important to keep in mind that there are no trivial changes. If we add a $resolved method that is synchronous, experience has shown that developers who are not vell versed in async ops will attempt to use this in lieu of the asynchronous $loaded, because it's "simpler." Thus, it has a significant cost. 

I'll want to see a pretty strong case for this before I'll greenlight having multiple, nuanced ways to accomplish the same goal.

**Comment by soumak77** [comment]
Just to be clear, `$resolved` is not a synchronous version of `$loaded`, nor is it a promise.  It is a boolean value which returns the current status of the $loaded promise.  This means that it returns false initially and then starts returning true once the `$loaded` promise has completed.  The use cases primarily focus around easier data binding with the `$firebaseObject` and `$firebaseArray`.

Per the ng-resource docs:
> $resolved: true after first server interaction is completed (either with success or rejection), false before that. Knowing if the Resource has been resolved is useful in data-binding.


The most straight forward use case is the one I already provided:
```javascript
$scope.obj = $firebaseObject(...);
```
```html
<div ng-if="obj.$resolved()">
  <!-- html which depends on obj being populated with data -->
</div>
```

The mirror example using ng-resource would be:
```javascript
$scope.obj = $resource(...);
```
```html
<div ng-if="obj.$resolved">
  <!-- html which depends on obj being populated with data -->
</div>
```

Often times when dealing with complex UI, there are cases when you need the data to be available before a directive or other HTML can be rendered properly.  Attempting to render the data before it is available cases the app to malfunction.  This is fixed by adding an `ng-if` attribute to wait to render the html until the data is available.

Ideally `$resolved` would be a property and not a function in firebase, like it is in ng-resource, but I couldn't get the solution working properly as a property (might just be some firebase specific issue I couldn't figure out).

**Comment by soumak77** [comment]
@katowulf any updates on this?

**Comment by soumak77** [comment]
One common example would be to show a loading symbol while the data is being retrieved:
```html
<div ng-show="obj.$resolved">
  <!-- show data -->
</div>
<div ng-show="!obj.$resolved">
  <!-- show loading symbol -->
</div>

**Comment by soumak77** [comment]
Here is a [live example](http://base-apps.github.io/angular-base-apps/templates/angular-firebase-template/#/) where a loading symbol is shown while downloading the content.  Let me know if you are leaning towards including this and I'll update my PR to convert `$resolved` into a property instead of a function.

**Comment by katowulf** [comment]
Please update the PR to use a property and we'll merge. Thanks for your patience and your contribution!

**Comment by soumak77** [comment]
@katowulf done

**Comment by katowulf** [comment]
Looks like there was no "fix" comment in #837; it fixes this.
